nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/src/main/java/org/apache/hive/streaming/RecordsEOFException.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hive.streaming;
+
+/**
+ * This is a "marker class" used by the HiveRecordWriter to indicate there are no more records in the input stream.
+ * It is used by PutHive3Streaming to determine that all records have been written to transaction(s). IT extencs
+ */
+public class RecordsEOFException extends SerializationError {
```

Comment:
Hive's exception hierarchy looks a bit strange to me (`SerializationError` which is an `Exception` in fact) and we need to pass a dummy `new Exception()` in the constructor.
Would not it be better to descend from `RuntimeException` instead?

Reply:
Probably, I'm not even sure why I picked that (but I did have to choose the same package name in order to use the constructor I think). Will change to RuntimeException

Reply:
My idea was wrong, `RuntimeException`-s are handled differently by the Hive client and the transaction would be marked dead:
https://github.com/apache/hive/blob/bfb55568bcaa8e91112f4e5b983edcc9dcd2844c/streaming/src/java/org/apache/hive/streaming/HiveStreamingConnection.java#L814
Please put back the original `SerializationError`.

Reply:
That must've been why I picked that :) Will restore